site/app/controllers/admin/AutogradingConfigController.php

```diff
@@ -232,14 +233,47 @@ public function configUsedBy($config_path = null) {
     }
 
     /**
-     * @Route("/courses/{_semester}/{_course}/notebook_builder/{g_id}", methods={"GET"})
+     * @Route("/courses/{_semester}/{_course}/notebook_builder/{g_id}/{mode}", methods={"GET"})
```

Comment:
We can add a requirement on the final parameter to enforce it has to be a specific value:

```suggestion
     * @Route("/courses/{_semester}/{_course}/notebook_builder/{g_id}/{mode<new|edit>}", methods={"GET"})
```

See https://symfony.com/doc/current/routing.html#parameters-validation for documentation on this.

Reply:
Committed suggestion.